### PR TITLE
Handling JWT errors

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

**Comment by bhariharan** [comment]
Ready to merge now. I've added generic `Intent` broadcasts for `OAuth` errors too, so consuming apps can do more than just the default `Toast` message we display, without having to override the class.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java [hunk 1]
```diff
@@ -240,6 +243,8 @@ public void run() {
             }, t.getDuration());
             t.show();
         }
+        final Intent intent = new Intent(AUTHENTICATION_FAILED_INTENT);
```
Comment (csoon03) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java]: I was able to receive the intent! Just wanted to confirm before I report back to UX and PM, there's no way to differentiate the errors?

Comment (csoon03) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java]: Also, I was looking at UserAccountManager.USER_SWITCH_INTENT_ACTION but this seems to only be sent if the user is switching to another user that has logged in previously, are there no intents sent for users who have successfully completed the login flow?

Comment (huminzhi) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java]: So you will need to define and check the error description to differentiate..

File: libs/SalesforceSDK/res/values/sf__strings.xml [hunk 2]
```diff
@@ -8,6 +8,7 @@
 	<string name="sf__invalid_server_url">Please type in a valid URL starting with https://</string>
 	<string name="sf__access_token_revoked">Your access token has been revoked by the administrator. You will now be logged out.</string>
     <string name="sf__managed_app_error">Authentication only allowed from managed device.</string>
+	<string name="sf__jwt_authentication_error">JWT authentication error. Please try again.</string>
```
Comment (huminzhi) [inline libs/SalesforceSDK/res/values/sf__strings.xml]: you can redefine the error description here..

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java (hunk 1 above)
Comment (csoon03) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java]: but there's nothing passed to the Intent that I can check against?

Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java]: @csoon03 I can pass in `error` and `errorDesc` if you like. `error` is a `String` that represents the error like `access_denied`, for instance, and `errorDesc` is a human readable description of that. Typically, these fields are set for errors when the request fails that are returned from the server. If nothing is returned from the server, these fields might be empty. Does that work for you?

File: libs/SalesforceSDK/res/values/sf__strings.xml (hunk 2 above)
Comment (bhariharan) [inline libs/SalesforceSDK/res/values/sf__strings.xml]: Actually, you wouldn't modify this file. You would override the value for this key in your app's `strings.xml`.

Comment (csoon03) [inline libs/SalesforceSDK/res/values/sf__strings.xml]: Gotcha!

**Comment by bhariharan** [comment]
I worked with @csoon03 offline and tested the flow end-to-end. Seems to be working well. This should be good to go.

**Pull request merged**